modules/Router.js

```diff
@@ -91,34 +79,14 @@ const Router = React.createClass({
     let { history } = this.props
     const { routes, children } = this.props
 
-    if (isDeprecatedHistory(history)) {
-      history = this.wrapDeprecatedHistory(history)
-    }
-
     const transitionManager = createTransitionManager(
-      history, createRoutes(routes || children)
+      history,
+      createRoutes(routes || children)
     )
     const router = createRouterObject(history, transitionManager)
-    const routingHistory = createRoutingHistory(history, transitionManager)
-
-    return { history: routingHistory, transitionManager, router }
-  },
-
-  wrapDeprecatedHistory(history) {
-    const { parseQueryString, stringifyQuery } = this.props
-
-    let createHistory
-    if (history) {
-      warning(false, 'It appears you have provided a deprecated history object to `<Router/>`, please use a history provided by ' +
-                     'React Router with `import { browserHistory } from \'react-router\'` or `import { hashHistory } from \'react-router\'`. ' +
-                     'If you are using a custom history please create it with `useRouterHistory`, see http://tiny.cc/router-usinghistory for details.')
-      createHistory = () => history
-    } else {
-      warning(false, '`Router` no longer defaults the history prop to hash history. Please use the `hashHistory` singleton instead. http://tiny.cc/router-defaulthistory')
-      createHistory = createHashHistory
-    }
+    history = { ...history, ...transitionManager }
```

Comment:
Don't need the "routing history" any more. This function shouldn't return `history` at all.